### ISSUE TITLE
 Executable path is not absolute, ignoring: sh db-container-backup.sh

### ISSUE THREAD

**Issue by MajkelP** [ChristianLempa/scripts#4]
Thank you for your work Christian, you are doing a great work. 
I would like use it above script in my Synology NAS in Docker. I have  Microsoft SQL2019 database and I need it to do a schedule backup. I fallowed your channel on YouTube and tutorials. Unfortunately I am not very familiar wit some Linux errors and it is hard to me to trouble shooting them. 
Maybe I describe what I have done: 
1. I created folder called "backup" via DSM in GUI. Above folder is on the volume1.Do I need set up any permission or leave it as is default. To this folder I copied three files: 
db-container-backup.service,  db-container-backup.sh, db-container-backup.timer
2. According to your instruction I copied two files:
db-container-backup.service,   
db-container-backup.timer
to /etc/systemd/system
Do I need give them any permissions? (root, user)?
After that I run
sudo systemctl enable db-container-backup
and checked if does it work:
sudo systemctl status db-container-backup
and I saw:

`Jan 02 15:58:42 DS918p systemd[1]: [/etc/systemd/system/db-container-backup.service:7] Executable path is not absolute, ignoring: sh db-container-backup.sh`
`Jan 02 15:58:42 DS918p systemd[1]: db-container-backup.service lacks both ExecStart= and ExecStop= setting. Refusing.`
`Jan 02 16:04:22 DS918p systemd[1]: [/etc/systemd/system/db-container-backup.service:7] Executable path is not absolute, ignoring: sh db-container-backup.sh`
`Jan 02 16:04:22 DS918p systemd[1]: db-container-backup.service lacks both ExecStart= and ExecStop= setting. Refusing.`
`Jan 02 16:04:55 DS918p systemd[1]: [/etc/systemd/system/db-container-backup.service:7] Executable path is not absolute, ignoring: sh db-container-backup.sh`
`Jan 02 16:04:55 DS918p systemd[1]: db-container-backup.service lacks both ExecStart= and ExecStop= setting. Refusing.`
`Jan 02 16:07:04 DS918p systemd[1]: [/etc/systemd/system/db-container-backup.service:7] Executable path is not absolute, ignoring: sh db-container-backup.sh`
`Jan 02 16:07:04 DS918p systemd[1]: db-container-backup.service lacks both ExecStart= and ExecStop= setting. Refusing.`
`Jan 02 16:13:32 DS918p systemd[1]: [/etc/systemd/system/db-container-backup.service:7] Executable path is not absolute, ignoring: sh /volume1/backup/db-container-backup.sh`
`Jan 02 16:13:32 DS918p systemd[1]: db-container-backup.service lacks both ExecStart= and ExecStop= setting. Refusing.`
`Jan 02 16:18:53 DS918p systemd[1]: [/etc/systemd/system/db-container-backup.service:7] Executable path is not absolute, ignoring: sh /volume1/backup/db-container-backup.sh`
`Jan 02 16:18:53 DS918p systemd[1]: db-container-backup.service lacks both ExecStart= and ExecStop= setting. Refusing.`

I am not quite sure what: "Executable path is not absolute," means and where I made a mistake.
I also understand that "db-container-backup.sh" must be executable so I run chmod +x db-container-backup.sh.

I do not have any other database - only one mssql. I understand that I can leave the script as it is. 

I amended only one line in the db-container-backup.sh. :
`
DAYS=2
BACKUPDIR=/volume1/backup`

and in the db-container-backup.service I changed User name.

`[Unit]`
`Description=DB Container Backup Script`
`Wants=db-container-backup.timer`
`[Service]`
`Type=simple`
`ExecStart=sh db-container-backup.sh`
`User=MyUserName`
`[Install] `
`WantedBy=default.target `

I also would like to ask how to change the time of backup and be more specific, daily at 10PM?  

`[Unit]`
`Description=DB Container Backup Daily Job`
`[Timer]`
`OnCalendar=daily`
`Persistent=true`
`Unit=db-container-backup.service`
`[Install]`
`WantedBy=timers.target`

If you or somebody could help I would be very greatfull. 
Thank you,

**Comment by MajkelP** [comment]
I think I have partially solved the problem.
In the Description=DB Container Backup Script file I changed 
`ExecStart=/bin/bash -c /volume1/backup/db-container-backup.sh`

but still having the error:

Myusername@DS918p:/volume1/backup$ sudo systemctl status db-container-backup
● db-container-backup.service - DB Container Backup Script
   Loaded: loaded (/etc/systemd/system/db-container-backup.service; enabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Tue 2023-01-03 09:48:53 CET; 12s ago
  Process: 21432 ExecStart=/bin/bash -c /volume1/backup/db-container-backup.sh (code=exited, status=126)
 Main PID: 21432 (code=exited, status=126)

Jan 03 09:48:53 DS918p systemd[1]: Started DB Container Backup Script.
Jan 03 09:48:53 DS918p systemd[1]: Starting DB Container Backup Script...
Jan 03 09:48:53 DS918p systemd[1]: db-container-backup.service: main process exited, code=exited, status=126/n/a
Jan 03 09:48:53 DS918p systemd[1]: Unit db-container-backup.service entered failed state.
Jan 03 09:48:53 DS918p systemd[1]: db-container-backup.service failed.

**Comment by ChristianLempa** [comment]
@MajkelP I'm not sure if that script will work for Microsoft SQL, as it is different from a MySQL/MariaDB Syntax.

**Comment by MajkelP** [comment]
Hi Christian, 
thank you for your answer, of course, I was blind, I mixed mysql with mssql! oh... :)
But I was able to manage to do some automation on my Synology. I am able to do a backup MSSQL database out the Docker container to the host. I am using something like task scheduler in DSM, which runs batch script on daily basis.
Now I need to do some script to delete backups database older than 30 days form my hard drive. 
Thank you to tip about Duplicati.
